crates/topos-api/src/graphql/checkpoint.rs

```diff
@@ -0,0 +1,43 @@
+use async_graphql::InputObject;
+use serde::{Deserialize, Serialize};
+
+use topos_uci::{self, SUBNET_ID_LENGTH};
+
+#[derive(Debug, Default, Serialize, Deserialize, InputObject)]
+pub struct CertificateId {
+    pub value: String,
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, InputObject)]
+pub struct InputSubnetId {
+    pub value: String,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, InputObject)]
+pub struct SourceStreamPosition {
+    pub source_subnet_id: InputSubnetId,
+    pub position: u64,
+    pub certificate_id: Option<CertificateId>,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, InputObject)]
+pub struct SourceCheckpoint {
+    pub source_subnet_ids: Vec<InputSubnetId>,
+    pub positions: Vec<SourceStreamPosition>,
+}
+
+impl InputSubnetId {
+    pub fn to_uci_subnet_id(self) -> topos_uci::SubnetId {
+        // Remove the leading "0x" from the hexadecimal string
+        let hex_string = self.value.trim_start_matches("0x");
+
+        // Decode the hexadecimal string into a vector of bytes
+        let bytes = hex::decode(hex_string).expect("Failed to decode hexadecimal string");
```

Comment:
I believe we should not use expect in production code. If it could fail, function should return error.